pkg/controller/core/localqueue_controller.go

```diff
@@ -74,20 +78,36 @@ func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling LocalQueue")
 
-	// Shallow copy enough for now.
-	oldStatus := queueObj.Status
-
-	pending, err := r.queues.PendingWorkloads(&queueObj)
+	var cq kueue.ClusterQueue
+	queueIsInactiveMsg := "Can't submit new workloads to clusterQueue"
+	err := r.client.Get(ctx, client.ObjectKey{Name: string(queueObj.Spec.ClusterQueue)}, &cq)
 	if err != nil {
-		r.log.Error(err, "Failed to retrieve localQueue status")
-		return ctrl.Result{}, err
-	}
-
-	queueObj.Status.PendingWorkloads = pending
-	queueObj.Status.AdmittedWorkloads = r.cache.AdmittedWorkloadsInLocalQueue(&queueObj)
-	if !equality.Semantic.DeepEqual(oldStatus, queueObj.Status) {
-		err := r.client.Status().Update(ctx, &queueObj)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		if apierrors.IsNotFound(err) {
+			err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, "ClusterQueueDoesNotExist", queueIsInactiveMsg)
+			if err != nil {
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		} else {
+			r.log.Error(err, "Failed to get a clusterQueue that localQueue references",
+				"clusterQueue", klog.KRef("", string(queueObj.Spec.ClusterQueue)))
+			// Update status only.
+			err = r.UpdateStatusIfChanged(ctx, &queueObj, "", "", "")
+			if err != nil {
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+	} else {
```

Comment:
avoid else by returning above when there is nothing else to do.

Reply:
Done.